farmdata2_modules/fd2_tabs/fd2_example/ui.html

```diff
@@ -176,11 +213,56 @@
 
         waitingOnLogs: false,
         seedingLogs: [],
+
+        regularExp: null,
+        selectedRegOpt: null,
+        regexInputVal: null,
+        regexMatchVal: false,
+        disabledMap: {
+            "DropdownWithAllComponent": false,
+            "RegexInputComponent": false,
+            "DateSelectionComponent": false,
+            "DateRangeSelectionComponent": false,
+        }
     },
     methods: {
+        checkDisabled(key) {
+            if(this.disabledMap[key]) {
+                return "Enable"
+            } else {
+                return "Disable"
+            }
+        },
+        makeDisabled(key) {
+            this.disabledMap[key] = !this.disabledMap[key];
+        },
         cropChange(newCrop) {
             this.pickedCrop = newCrop
         },
+        setNewRegexMatch(newMatch){
+            this.regexMatchVal = newMatch
+        },
+        setNewRegexVal(newVal){
+            this.regexInputVal = newVal 
+        },
+        resetRegexVal(){
+            this.regexInputVal = '0' 
+        },
+        pickRegExp(newOption) {
```

Comment:
I would suggest using a map to hold the regex and then pass the key to this function from the @click event.   That way there is no if statement in this function and the calls to it from the two buttons are consistent.